riscv/processor.cc

```diff
@@ -832,8 +832,13 @@ const char* processor_t::get_privilege_string()
 
 void processor_t::enter_debug_mode(uint8_t cause)
 {
+  const bool has_zicfilp = extension_enabled(EXT_ZICFILP);
   state.debug_mode = true;
-  state.dcsr->write_cause_and_prv(cause, state.prv, state.v);
+  state.dcsr->update_fields(cause, state.prv, state.v,
+                            (has_zicfilp ? state.elp : elp_t::NO_LP_EXPECTED));
```

Comment:
If extension is not enabled then state.elp should be just NO_LP_EXPECTED - so can remove the has_zicfilp based select.

Reply:
Good point. Fixed.